designs/kwok-provider.md

```diff
@@ -0,0 +1,51 @@
+# Karpenter KwoK Cloud Provider - v0.34+
+
+The kubernetes-sigs/karpenter repo uses a simulated kubernetes environment to test code changes. Users must use a cloud provider of their choice to test their changes, which can be costly and creates an unreasonable barrier to entry for contributors. Related issue: https://github.com/kubernetes-sigs/karpenter/issues/895.
+
+Creating a Cloud Provider neutral implementation of Karpenter within Karpenter-core requires creating a set of CP neutral examples of instance types and pricing that captures common tenets across CPs. 
+
+This is meant to be a living document to in-depth go through the decisions on how kubernetes-sigs/karpenter models its fake instance types. 
+
+## Proposal
+
+### Instance Types
+The full set of instance types for the KwoK Cloud Provider should be at least the cartesian product of the following dimensions, generating 60 instance types across 4 zones, totaling `12 * 3 * 2 = 64` in-memory Cloud Provider instance types.
+
+```md
+1. Instance Type Name: (family)-(size)
+    a. size: 1x, 2x, 4x, 9x, 16x, 32x, 48x, 64x, 100x, 128x, 192x, 256x
+    b. family: c (cpu), s (standard), m (memory)
+        i. c ---> 1 vCPU : 2 GiB
+        ii. s --> 1 vCPU : 4 GiB
+        iii. m -> 1 vCPU : 8 GiB
+2. Capacity Type: [on-demand, spot]
```

Comment:
Totally agree. This is a good callout. It would be super useful to have a shared list here. 

The kwok provider should definitely be able to select by vCPU and GPU (when GPUs are implemented)